maxim/apis/maxim_apis.py

```
@@ -189,53 +178,47 @@ def __make_network_call_with_retry(
                     )
                     time.sleep(delay)
                     continue
                 else:
                     scribe().error(
-                        f"[MaximSDK] Failed to establish connection after {max_retries + 1} attempts. Last error: {e}"
+                        f"[MaximSDK] Connection pool exhausted after {max_pool_retries} pool retries. Last error: {e}"
                     )
                     raise Exception(
-                        f"Connection failed after {max_retries + 1} attempts: {e}"
+                        f"Connection pool exhausted after {max_pool_retries} retries: {e}"
                     ) from e
```

Ruff (0.11.9)

Create your own exception

(TRY002)